### PR TITLE
build: auto allocate

### PR DESCRIPTION
**Pull request by Schlagonia**
## Description

- Add an `auto_allocate` flag to the be set by the DEBT_MANAGER to automatically push all available debt to the first strategy in the queue during every deposit and mint

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: contracts/VaultV3.vy [hunk 1]
```diff
@@ -1048,12 +1061,14 @@ def _update_debt(strategy: address, target_debt: uint256, max_loss: uint256) ->
         withdrawable: uint256 = IStrategy(strategy).convertToAssets(
             IStrategy(strategy).maxRedeem(self)
         )
-        assert withdrawable != 0, "nothing to withdraw"
```
Comment (wavey0x) [inline contracts/VaultV3.vy:1051]: is this related?

**Review by wavey0x — APPROVED** [review]

File: contracts/VaultV3.vy (hunk 1 above)
Comment (Schlagonia) [reply contracts/VaultV3.vy:1051]: I removed the asserts when increasing debt to not revert if it cant be done. So deposits wont revert if auto allocating.

So i also removed the assert with removing debt for continuity of how the function works both ways.

Not sure if it makes sense or not TBH.

**Review by fp-crypto — CHANGES_REQUESTED** [review]
docs and spec

File: contracts/VaultV3.vy [hunk 2]
```diff
@@ -1094,12 +1109,18 @@ def _update_debt(strategy: address, target_debt: uint256, max_loss: uint256) ->
...
+        max_debt: uint256 = self.strategies[strategy].max_debt
+        if new_debt > max_debt:
+            new_debt = max_debt
+            # Possible for current to be greater than max from reports.
+            if new_debt < current_debt:
+                return current_debt
 
         # Vault is increasing debt with the strategy by sending more funds.
         max_deposit: uint256 = IStrategy(strategy).maxDeposit(self)
-        assert max_deposit != 0, "nothing to deposit"
```
[the remark this answers is not in the document]
Comment (Schlagonia) [reply contracts/VaultV3.vy:1102]: require a state change for the txn to not revert

**Comment by Schlagonia** [comment]
> docs and spec

https://github.com/yearn/yearn-vaults-v3/pull/203/commits/1c2b05843816bbebaa665c92edb1eccf9841f7ce

**Review by fp-crypto — APPROVED** [review]
LGTM, pending all tests passing
